tasks/RedHat/generic/assert-tuned.yml

```diff
@@ -6,22 +6,15 @@
 - name: Report tuned version
   debug:
     msg: "INFO: The installed version of package tuned is: {{ ansible_facts.packages['tuned'][0].version }}"
-  ignore_errors: yes
-  when: sap_hana_preconfigure_use_tuned or
-        sap_hana_preconfigure_assert_all_config
-
-- name: Get status of tuned
-  command: systemctl status tuned
-  register: tuned_status
-  ignore_errors: yes
-  changed_when: no
-  when: sap_hana_preconfigure_use_tuned or
-        sap_hana_preconfigure_assert_all_config
+  when: "'tuned' in ansible_facts.packages and
+        (sap_hana_preconfigure_use_tuned or
+           sap_hana_preconfigure_assert_all_config
+        )"
 
 - name: Assert that tuned is enabled
   assert:
     that:
-      - "'/usr/lib/systemd/system/tuned.service; enabled' in tuned_status.stdout"
+      - ansible_facts.services['tuned.service']['status'] == 'enabled'
```

Comment:
Are you guaranteed that 1) `ansible_facts.services` is set i.e. you have used `service_facts` and 2) `tuned.service` is in `services`?  If not, then maybe add a `when`
```yaml
when:
  - ansible_facts.services | d(false)
  - "'tuned.service' in ansible_facts.services
```

Reply:
1) I am collecting service_facts in higher level assert-configuration.yml
2) If package tuned is not installed, then might be no "tuned.service"  in ansible_facts.services (except if the service is running but the package had been removed).
Checking...

Reply:
typically, if you add a package, the RPM spec file will enable and start the service - it will do
```
/usr/bin/systemctl --no-reload preset NAME.service
```
similarly, if you remove a package, the RPM spec file will stop and disable the service - it will do
```
/usr/bin/systemctl --no-reload disable --now NAME.service
```
however, you will need to reload `service_facts` in order to get the state, or use some other mechanism like using the `systemctl` command directly to check the status of the service

Reply:
I will add one more assertion, for the presence of the tuned service before this task. Same for sap-netweaver-preconfigure.